src/uniswap-summaries.md

```diff
@@ -3155,6 +3155,154 @@ module SOLIDITY-UNISWAP-SWAP-SUMMARY
        <current-function> swap </current-function>
     requires Vto =/=MInt {S[token0] orDefault default(address)}:>MInt{160} andBool Vto =/=MInt {S[token1] orDefault default(address)}:>MInt{160} [priority(40)]
 
+  // End of first if statement to second if condition evaluation
+  rule <k> if ( amount1Out > 0 ) iERC20 ( vidToken1 , .TypedVals ) . transfer ( to , amount1Out , .TypedVals ) ; Ss:Statements => if ( v( Va1 >uMInt 0p256, bool) ) iERC20 ( vidToken1 , .TypedVals ) . transfer ( to , amount1Out , .TypedVals ) ; ~> Ss ...</k>
+       <summarize> true </summarize>
+       <env>... amount1Out |-> var(Ia1, uint256) ...</env>
+       <store> _ [ Ia1 <- Va1 ] </store>
+       <this-type> uniswapV2Pair </this-type>
+       <current-function> swap </current-function> [priority(40)]
+
+  // Second if condition evaluated to true to transfer call
+  rule <k> if ( v( true, bool) ) iERC20 ( vidToken1 , .TypedVals ) . transfer ( to , amount1Out , .TypedVals ) ; => v ( Vv1 , iERC20 ) . transfer ( v ( Vto , address ) , v ( Va1 , uint256 ) , .TypedVals ) ~> freezerExpressionStatement ( ) ...</k>
+       <summarize> true </summarize>
+       <env>... (amount1Out |-> var(Ia1, uint256))
+                (to |-> var(Ito, address))
+                (vidToken1 |-> var(Iv1, address))
+       ...</env>
+       <store> _ [ Ia1 <- Va1 ] [ Ito <- Vto ] [ Iv1 <- Vv1 ] </store>
+       <this-type> uniswapV2Pair </this-type>
+       <current-function> swap </current-function> [priority(40)]
+
+  // After if conditions: compute params and address for first balanceOf call
+  rule <k> balance0 = iERC20 ( vidToken0 , .TypedVals ) . balanceOf ( address ( this , .TypedVals ) , .TypedVals ) ; Ss:Statements => v ( Vv0 , iERC20 ) . balanceOf ( v ( THIS , address ) , .TypedVals ) ~> freezerAssignment ( balance0 ) ~> freezerExpressionStatement ( ) ~> Ss...</k>
+       <summarize> true </summarize>
+       <env>... (vidToken0 |-> var(Iv0, address)) ...</env>
+       <store> _ [ Iv0 <- Vv0 ] </store>
+       <this> THIS:MInt{160} </this>
```

Comment:
There's really a need for this cast?

Reply:
It is not a cast, it is just matching only when THIS is of the indicated type. Not necessary, since THIS will always be of that type, just making the type of THIS immediately obvious.